### ISSUE TITLE
Can I use any encrypt decryption on video while cacheing or playing,

### ISSUE THREAD

**Issue by shailesh242121** [danikula/AndroidVideoCache#71]
Hi,
  This is not a bug/ issue, But I wana use a encryption and decryption on video while buffering or caching.
Is there any possibility or have some library for doing this. can you suggest any one.

**Comment by danikula** [comment]
@shailesh242121 I think it is possible to add some basic encryption. But it will have few limitations:
* it should work quickly
* encrypted video **must** have same size. Without it library will not work at all.

So it is possible to implement some kind of primitive encryption based on XOR or smth like this. If you interested in implementation of this feature, let's me know.

**Comment by ZohaibGhafoor** [comment]
any solution on encryption @danikula ? i need this feature in my app.

**Comment by danikula** [comment]
I started to implement it year ago, but I've not finished it yet. I can push draft of implementation to separate branch and you if want can finish it.

**Comment by ZohaibGhafoor** [comment]
Yes i will. but i am not as professional as you are. If you give me any reference of your logic implementation and about encryption to read and understand so it will be better to work around on it.

**Comment by danikula** [comment]
I pushed my draft implementation to [encrypt](https://github.com/danikula/AndroidVideoCache/commits/encrypt) branch. `Cipher` is responsible for encryption/decryption. Any you need to do is write fast `Cipher` implementation and use `CipherAwareCacheWrapper`

**Comment by m7mdra** [comment]
@danikula any updates ?  thanks for the effort
